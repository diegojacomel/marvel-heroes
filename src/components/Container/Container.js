// Modules
import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
    width: calc(${props => props.theme.spacing.full} - ${props => props.theme.spacing.lg});
    padding: ${props => props.theme.spacing.none} ${props => props.theme.spacing.ssm};
`

const ContainerWrapper = ({ children }) => (
    <Wrapper>
        {children}
    </Wrapper>
)

function Container({ children }) {
    return (
        <ContainerWrapper id="testContainer">
            {children}
        </ContainerWrapper>
    )
}

export default Container;