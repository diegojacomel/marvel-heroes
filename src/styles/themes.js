// Themes
const primaryTheme = {
    colors: {
        primary: '#003366',
        secondary: '#000000',
        warning: '#000000',
        danger: '#d30d30',
        info: '#006633',
        default: '#dddddd',
        grayLight: '#f2f2f2',
        grayMedium: '#dddddd',
        grayDark: '#333333',
        black: '#000000',
        white: '#ffffff',
        hover: {
            primary: '#115394',
            secondary: '#000000',
            warning: '#000000',
            danger: '#d30d30',
            info: '#006633',
            default: '#dddddd',
            grayLight: '#f2f2f2',
            grayMedium: '#dddddd',
            grayDark: '#333333',
            black: '#000000',
            white: '#ffffff'
        }
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '75px',
        ggt: '100px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    }
}

export {
    primaryTheme
}