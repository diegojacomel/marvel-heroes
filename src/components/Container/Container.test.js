// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/themes'; 

// Component
import Container from './Container';

describe('Logo', () => {
    const wrapper = shallow(<Container />);

    test('It exist', () => {
        expect(wrapper.find('#testContainer')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testContainer');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testContainer')).toHaveDisplayName('ContainerWrapper');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testContainer')).toMatchSelector('ContainerWrapper');
    })

    test('Have props', () => {
        expect(wrapper.find('#testContainer')).toHaveProp('children');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Container>
                </Container>
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})