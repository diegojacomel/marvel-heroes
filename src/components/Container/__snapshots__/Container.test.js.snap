// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Render correctly 1`] = `
<div
  className="sc-bdVaJa ftubKR"
/>
`;
